src/System.Memory/src/System/ReadOnlySpan.cs

```diff
@@ -220,35 +220,78 @@ public void CopyTo(Span<T> destination)
         /// <param name="destination">The span to copy items into.</param>
         public bool TryCopyTo(Span<T> destination)
```

Comment:
Should we put the logic of this method (or the whole TryCopyTo method) into a helper function to avoid code duplication?